src/plugins/navigation/mutations/createNavigationTree.js

```diff
@@ -0,0 +1,60 @@
+import Random from "@reactioncommerce/random";
+import decodeNavigationTreeItemIds from "../util/decodeNavigationTreeItemIds.js";
+import setDefaultsForNavigationTreeItems from "../util/setDefaultsForNavigationTreeItems.js";
+import { NavigationTree as NavigationTreeSchema } from "../simpleSchemas.js";
+
+/**
+ * @method createNavigationTree
+ * @summary Creates a navigation tree
+ * @param {Object} context An object containing the per-request state
+ * @param {Object} input Input of createNavigationTree mutation
+ * @param {String} input.navigationTreeId ID of navigation tree to update
+ * @param {String} input.shopId Shop ID of navigation tree
+ * @param {Object} input.navigationTree Navigation tree object to update
+ * @returns {Promise<Object>} Created navigation tree
+ */
+export default async function createNavigationTree(context, input) {
+  const { collections } = context;
+  const { NavigationTrees } = collections;
+  const { shopId, navigationTree } = input;
+
+  const {
+    shouldNavigationTreeItemsBeAdminOnly,
+    shouldNavigationTreeItemsBePubliclyVisible,
+    shouldNavigationTreeItemsBeSecondaryNavOnly
+  } = await context.queries.appSettings(context, shopId);
+
+  // Navigation item visibility defaults from settings
+  const visibilityDefaults = {
+    shouldNavigationTreeItemsBeAdminOnly,
+    shouldNavigationTreeItemsBePubliclyVisible,
+    shouldNavigationTreeItemsBeSecondaryNavOnly
+  };
+
+  // Set default values for navigation tree draft items before validation.
+  // isVisible, isPrivate, and isSecondary are optional in the GraphQL schema,
+  // but are required by SimpleSchema. This reduces the input payload size by not
+  // needing to include values that aren't explicitly set.
+  const navigationTreeData = {
+    ...navigationTree,
+    shopId,
+    _id: Random.id(),
+    draftItems: setDefaultsForNavigationTreeItems(navigationTree.draftItems, visibilityDefaults)
+  };
+
+  // Validate the navigation tree with the defaults set
+  NavigationTreeSchema.validate(navigationTreeData);
+  const { draftItems } = navigationTreeData;
+
+  await context.validatePermissions("reaction:navigationTrees", "create", { shopId, legacyRoles: ["core"] });
```

Comment:
@kieckhafer What should this permission name be now? No `legacyRoles` needed, right?

Reply:
Correct, no legacy roles needed:

```suggestion
  await context.validatePermissions("reaction:legacy:navigationTrees", "create", { shopId });
```

Reply:
thanks, @aldeed, and @kieckhafer for helping me with this, `roles` and `permission` are a bit tricky to understand at least for me. 

I would like to help you guys with the documentation if it's not already documented.

Reply:
👍 We do have some documentation written up internally at the moment, it will be publicized with the official 3.0 launch but we are still working through some of the functionality so didn't want to publish it publicly yet knowing it's going to change. 

But we agree, it's tricky!

Reply:
awesome, I am very excited about the v3 launch and hope that adoption will be more and complains will be far less. More importantly, the developer experience will be much better now.

Reply:
@kieckhafer can you not invite some contributors for writing documentation on their own parts of code? A bit like how component library have documentation next to the component in the folder hierarchy.